homeassistant/components/binary_sensor/zha.py

```diff
@@ -0,0 +1,80 @@
+"""
+Binary sensors on Zigbee Home Automation networks.
+
+For more details on this platform, please refer to the documentation
+at https://home-assistant.io/components/binary_sensor.zha/
+"""
+import asyncio
+import logging
+
+from homeassistant.components.binary_sensor import DOMAIN, BinarySensorDevice
+from homeassistant.components import zha
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['zha']
+
+CLASS_MAPPING = {
+    0x000d: 'motion',
+    0x0015: 'opening',
+    0x0028: 'smoke',
+    0x002a: 'moisture',
+    0x002b: 'gas',
+    0x002d: 'vibration',
+}
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
+    """Setup Zigbee Home Automation binary sensors."""
+    if discovery_info is None:
+        return
+
+    clusters = discovery_info['clusters']
+
+    device_class = None
+    cluster = [c for c in clusters if c.cluster_id == 0x0500][0]
```

Comment:
I think I want to get this constant out anyways. The structure of the code is also kinda gross.